xml_struct_derive/src/lib.rs

```diff
@@ -0,0 +1,185 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+mod properties;
+mod serialize;
+
+use syn::{parse_macro_input, DeriveInput};
+
+pub(crate) use properties::*;
+
+use crate::serialize::{write_serialize_impl_for_enum, write_serialize_impl_for_struct};
+
+const MACRO_ATTRIBUTE: &str = "xml_struct";
```

Comment:
I mostly defined it here because of its proximity to the `proc_macro_derive`, since it needs to match the value under `attributes`. I can make that relationship explicit, at least.